examples/python/cancellation/README.md

```diff
@@ -0,0 +1,183 @@
+### Cancelling RPCs
+
+RPCs may be cancelled by both the client and the server.
+
+#### The Example
+
+In the example, we implement a silly algorithm. We search for bytestrings whose
+hashes are similar to a given search string. For example, say we're looking for
+the string "doctor". Our algorithm may `JrqhZVkTDoctYrUlXDbL6pfYQHU=` or
+`RC9/7mlM3ldy4TdoctOc6WzYbO4=`. This is a brute force algorithm, so the server
+performing the search must be conscious the resources it allows to each client
+and each client must be conscientious of the resources demanded of the server.
+
+In particular, we ensure that client processes cancel the stream explicitly
+before terminating and we ensure the server cancels RPCs that have gone on longer
+than a certain number of iterations.
+
+#### Cancellation on the Client Side
+
+A client may cancel an RPC for several reasons. Perhaps the data it requested
+has been made irrelevant. Perhaps you, as the client, want to be a good citizen
+of the server and are conserving compute resources.
+
+##### Cancelling a Server-Side Unary RPC from the Client
+
+The default RPC methods on a stub will simply return the result of an RPC.
+
+```python
+>>> stub = hash_name_pb2_grpc.HashFinderStub(channel)
+>>> stub.Find(hash_name_pb2.HashNameRequest(desired_name=name))
+<hash_name_pb2.HashNameResponse object at 0x7fe2eb8ce2d0>
+```
+
+But you may use the `future()` method to receive an instance of `grpc.Future`.
+This interface allows you to wait on a response with a timeout, add a callback
+to be executed when the RPC completes, or to cancel the RPC before it has
+completed.
+
+In the example, we use this interface to cancel our in-progress RPC when the
+user interrupts the process with ctrl-c.
```

Comment:
The `signal` mechanism is not a necessary step for cancellation. I think a simple `future.cancel()` is capable of convey the idea of cancelling an unary request.

Reply:
Discussed offline. If you have a better way to motivate why you would cancel on the client side, I'm more than happy to give it a shot.